test/e2e/cluster_extension_install_test.go

```diff
@@ -322,6 +257,81 @@ func TestClusterExtensionInstallSuccessorVersion(t *testing.T) {
 	}, pollDuration, pollInterval)
 }
 
+func TestClusterExtensionInstallReResolvesWhenNewCatalog(t *testing.T) {
+	t.Log("When a cluster extension is installed from a catalog")
+	t.Log("It resolves again when a new catalog is available")
+
+	// Tag the image with the new tag
+	baseImage := "localhost:30000/e2e/test-catalog:v1"
+	var err error
+	err = crane.Tag(baseImage, "latest", crane.Insecure)
+	require.NoError(t, err)
+
+	// create a test-catalog with latest image tag
+	extensionCatalog, err := createTestCatalog(context.Background(), testCatalogName, os.Getenv(testLatestCatalogRefEnvVar))
+	require.NoError(t, err)
+	clusterExtensionName := fmt.Sprintf("clusterextension-%s", rand.String(8))
+	clusterExtension := &ocv1alpha1.ClusterExtension{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterExtensionName,
+		},
+	}
+	defer testCleanup(t, extensionCatalog, clusterExtension)
+	defer getArtifactsOutput(t)
+
+	pkgName := "prometheus"
+	clusterExtension.Spec = ocv1alpha1.ClusterExtensionSpec{
+		PackageName:      pkgName,
+		InstallNamespace: "default",
+	}
+	t.Log("It resolves the specified package with correct bundle path")
+	t.Log("By creating the ClusterExtension resource")
+	require.NoError(t, c.Create(context.Background(), clusterExtension))
+
+	t.Log("By reporting a successful resolution and bundle path")
+	require.EventuallyWithT(t, func(ct *assert.CollectT) {
+		assert.NoError(ct, c.Get(context.Background(), types.NamespacedName{Name: clusterExtension.Name}, clusterExtension))
+		assert.Len(ct, clusterExtension.Status.Conditions, len(conditionsets.ConditionTypes))
+		cond := apimeta.FindStatusCondition(clusterExtension.Status.Conditions, ocv1alpha1.TypeResolved)
+		if !assert.NotNil(ct, cond) {
+			return
+		}
+		assert.Equal(ct, metav1.ConditionTrue, cond.Status)
+		assert.Equal(ct, ocv1alpha1.ReasonSuccess, cond.Reason)
+		assert.Contains(ct, cond.Message, "resolved to")
+		assert.Equal(ct, &ocv1alpha1.BundleMetadata{Name: "prometheus-operator.1.2.0", Version: "1.2.0"}, clusterExtension.Status.ResolvedBundle)
+	}, pollDuration, pollInterval)
+
+	// update tag on test-catalog image with v2 image
+	t.Log("By creating an ClusterExtension catalog with the updated ClusterCatalog")
```

Comment:
This log message seems incorrecT?

```suggestion
	t.Log("By updating the catalog tag to point to the v2 catalog.")
```